### ISSUE TITLE
Cannot create asset change in p2sh transactions 

### ISSUE THREAD

**Issue by Tylerlhess** [RavenProject/Ravencoin#1174]
### Describe the issue
There is no way to define a change address for p2sh asset transfers (createrawtransaction transfer/transferwithmessage).

In the Transfer RPC it checks for and adds in the change to the transaction before sending it. With p2sh you have to use createrawtransaction RPCs to construct a transaction. As such you can only send the entire utxo or the unspent portion of the asset will be burned. (error code is generated as much and so it won't allow you to send anyways.) 

tx id = {'result': None, 'error': {'code': -26, 'message': '16: bad-tx-inputs-outputs-mismatch Bad Transaction - Assets would be burnt SQUAWKER'}, 'id': 13}

I could build another output transaction but this has to be to a different address to be contained in the same transaction json and building two raw transactions and combining them leads to decode errors (possibly from the p2sh address but I haven't looked into that).

So I can build the transactions, sign the transactions, but I can't fund the transactions (requires decode) and I can't combine the transactions (also requires decode). Sending transactions I think will work if I have perfect utxo change (needs further testing but the example seems to do this)

#### If so, please list the steps to reproduce below:
1. create a raw transaction using less of the asset than in in the input utxo
2. sign it
3. sending it will produce the error

### Expected behaviour
I would expect that there be options to define change_address and asset_change_address as part of createrawtransaction rpc as there is with the transfer RPC. Furthermore not sure why there is a createrawtransaction transfer and transferwithmessage when those are the same RPC

### Actual behaviour
Defining a p2sh asset transfer will burn the unused portion of the asset utxo so you can't send it as it errors on not having a change address in the transaction.
combining transactions also fails as there is an error decoding these raw transactions.

### What version of Ravencoin are you using?
4.7.0test3-fdov

**Comment by fdoving** [comment]
After discussions on Discord:
Do we agree this is just an issue when you try to send more than one output to the same address in the same transaction?

**Comment by Tylerlhess** [comment]
Yes but this is inconsistent with the non raw transactions where I can do this. I agree that you probably shouldn't be able to have the same address having multiple outputs in manual transactions. 

There should be a simple way to define change of assets and rvn to the same address. Currently it seems to be a one or the other as the createrawtransaction only allows send of rvn (option 1) or transferring an asset (options 2 and 3). Multiple assets transactions should be possible as its just a json dict you are providing for the assets so you could define the two different assets as distinct parts of the json object but it currently is a rvn or assets. 

I think we might be able to get around some of this using fundrawtransaction RPC which allows a change definition for RVN. If it also had the option to define an asset change address this would also solve my issue as I could send a UTXO to an address and then use the fundrawtransaction RPC to define the asset change location. Without really digging into the code I suspect this is how the transfer RPC works under the hood. Defining the transaction (createrawtransaction RPC) and then handling the change immediately afterwards (fundrawtransaction RPC) allowing each part to have the same address as an input and thus working around this whole issue. 


TL;DR; If we allow the fundrawtransaction RPC to define all change addresses for all leftover assets in addition to RVN this would solve this issue using the same pattern as pure rvn transactions. 

As a side note I would like to also be able to use fundrawtransaction defining the funds come from a specific address such as a p2sh address as it seems currently that I have to specify a rvn send and then use fundrawtransaction to remove the fee from that transaction. Granted I might be doing this all wrong as there are not very many resources/guides for doing RVN asset transactions let alone any that use raw transactions.

**Comment by Tylerlhess** [comment]
Closed issue as this is possible by creating your own raw transaction outside of core and having raven-qt sign it.
